wrappers/rust/icicle-core/src/polynomials/mod.rs

```diff
@@ -0,0 +1,707 @@
+                }
+            }
+        }
+
+        impl Add for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn add(self: Self, rhs: Self) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: add(self.handle, rhs.handle),
+                    }
+                }
+            }
+        }
+
+        impl AddAssign<&UnivariatePolynomial> for UnivariatePolynomial {
+            fn add_assign(&mut self, other: &UnivariatePolynomial) {
+                unsafe { add_inplace(self.handle, other.handle) };
+            }
+        }
+
+        impl Sub for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn sub(self: Self, rhs: Self) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: subtract(self.handle, rhs.handle),
+                    }
+                }
+            }
+        }
+
+        impl Mul for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn mul(self: Self, rhs: Self) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: multiply(self.handle, rhs.handle),
+                    }
+                }
+            }
+        }
+
+        // poly * scalar
+        impl Mul<&$field> for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn mul(self: Self, rhs: &$field) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: multiply_by_scalar(self.handle, rhs),
+                    }
+                }
+            }
+        }
+        // scalar * poly
+        impl Mul<&UnivariatePolynomial> for &$field {
+            type Output = UnivariatePolynomial;
+
+            fn mul(self, rhs: &UnivariatePolynomial) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: multiply_by_scalar(rhs.handle, self),
+                    }
+                }
+            }
+        }
+
+        impl Div for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn div(self: Self, rhs: Self) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: quotient(self.handle, rhs.handle),
+                    }
+                }
+            }
+        }
+
+        impl Rem for &UnivariatePolynomial {
+            type Output = UnivariatePolynomial;
+
+            fn rem(self: Self, rhs: Self) -> Self::Output {
+                unsafe {
+                    UnivariatePolynomial {
+                        handle: remainder(self.handle, rhs.handle),
+                    }
+                }
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! impl_polynomial_tests {
```

Comment:
I can try again but what is the motivation anyway?

Reply:
@yshekel 
1. It's easier to navigate through code when logically separated parts are separated into separate files
2. Once we commit to one way of doing something in one place of a codebase - we should keep it the same in all new parts of code

Reply:
Yes you are right. I tried to split it again and now I think I know how to do it and why I failed but it would take me ~2 days I etimate. Note that it won't affect user code and I am going to do it after documenting so unless you object, I would prefer doing after releasing V2.